src/Rules/Test.hs

```diff
@@ -15,47 +15,70 @@ import System.Environment
 -- TODO: clean up after testing
 testRules :: Rules ()
 testRules = do
+    root <- buildRootRules
+
+    -- | Using program shipped with testsuite to generate ghcconfig file.
+    root -/- ghcConfigProgPath ~> do
+        ghc             <- builderPath $ Ghc CompileHs Stage0
+        command [] ghc [ghcConfigHsPath, "-o" , root -/- ghcConfigProgPath]
+ 
+    -- | TODO : Use input test compiler and not just stage2 compiler.  
+    root -/- ghcConfigPath ~> do
+        ghcPath         <- needfile Stage1 ghc
+        need [ root -/- ghcConfigProgPath]
+        command [FileStdout $ root -/- ghcConfigPath] (root -/- ghcConfigProgPath)
+            [ ghcPath  ] 
```

Comment:
I think the variable `ghcPath` may be incorrect. In ghc's `testsuite/mk/boilerplate.mk` we pass the `$(TEST_HC)` as argument to `ghc-config`. However in the `"test"` rule, we use stage2 compiler as `TEST_HC`. I think here the consistency does matter.

Ref: https://github.com/ghc/ghc/blob/master/testsuite/mk/boilerplate.mk#L240

Reply:
Here also, the stage2 compiler is used only. We just build stage2 compiler in stage1 and this is why I have used stage1.

Reply:
In `need needTestBuilders` we ensure the stage2 ghc is ready before run test. The stage1 and stage2 compiler are often built in different ways (for example, with or without `-prof`). We are testing again the stage2 compiler, I think it does matter to use stage2 compiler to get these test settings (for example, `GhcWithInterpreter`, `GhcProfiled` and `GhcDebugged`).

> We just build stage2 compiler in stage1 and this is why I have used stage1.

I think we could `needBuilder $ Ghc CompileHs Stage2` in rule `root -/- ghcConfigPath`, before executing the `ghc-config` command.

Reply:
@sighingnow when I call needfile Stage1 ghc, I call the path of stage2 compiler only. Now, stage2 compiler is built in _build/stage1/bin/ghc. Hence, I am still pointing to stage2 compiler only.

Reply:
Yes you are correct. Thanks for the explanation. I just found that hadrain places the stage2 ghc inside the `stage1` folder....

Now this PR looks good to me.

Reply:
Thank you for the review @sighingnow!

Reply:
h I just found that hadrain places the stage2 ghc inside the stage1 folder....

Right, because it is _built_ by stage1. It is however indeed confusing at first, but it makes sense to me now.